src/lib.rs

```diff
@@ -378,10 +387,114 @@ impl Debug for Value {
             Value::ExternFunction(slot) => write!(fmt, "(extern fn {})", slot),
             Value::Unkown => write!(fmt, "(unkown)"),
             Value::Nil => write!(fmt, "(nil)"),
+            Value::Tuple(v) => write!(fmt, "({:?})", v),
         }
     }
 }
 
+mod op {
+    use super::Value;
+    use std::rc::Rc;
+
+    fn tuple_op(a: &Rc<Vec<Value>>, b: &Rc<Vec<Value>>, f: fn (&Value, &Value) -> Value) -> Value {
+        Value::Tuple(Rc::new(a.iter().zip(b.iter()).map(|(a, b)| f(a, b)).collect()))
+    }
+
+    pub fn neg(value: &Value) -> Value {
+        match value {
+            Value::Float(a) => Value::Float(-a),
+            Value::Int(a) => Value::Int(-a),
+            Value::Tuple(a) => Value::Tuple(Rc::new(a.iter().map(neg).collect())),
+            _ => Value::Nil,
+        }
+    }
+
+    pub fn not(value: &Value) -> Value {
+        match value {
+            Value::Bool(a) => Value::Bool(!a),
+            Value::Tuple(a) => Value::Tuple(Rc::new(a.iter().map(not).collect())),
```

Comment:
I would love to hear a use case for this. I'm not hard-opposed, but I don't see the value. Or is it just to match that the other operators all work tuples? In any case, I think I prefer Python's `all` and the like (which are easy to expand with more variants) instead of extending the `!`-operator.

Reply:
I just added it for consistency. Since now an operator applied to a tuple, is the same as applying the operator to all it's elements. It's not super usefull since it requires the tuple to be bools, but I don't think this is an alternative to `all`. I see it as a supliment. Note that it returns a tuple.

Reply:
Oh. That makes more sense actually.